### ISSUE TITLE
Test 'Unexpected Untagged Message Test' hang up in the PDS mode

### ISSUE THREAD

**Issue by svyatoslavnik** [ultraethernet/uet-ref-prov#50]
**How to reproduce:**
Run server and client applications with UET_PDS=PDS specified
```
sudo LD_LIBRARY_PATH=../libfabric/src/.libs UET_PDS=pds UET_IFNAME=brm ./uet server test 10.1.0.2
sudo LD_LIBRARY_PATH=../libfabric/src/.libs UET_PDS=pds UET_IFNAME=vm2 ./uet client test 10.1.0.1
```

**Syndrome:**

Test reach following errors
```
DBG: [PDS] PDC 1 [Rx 1] [PSN 141] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 141 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 142] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 142 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 143] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 143 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 144] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 144 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 141] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 141 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 142] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 142 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 143] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 143 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 144] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 144 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 141] [RUD_REQ/REQ_STD] - RX PACKET (1124)
[uet_pds.c:1492] WARN: [PDS] invalid PSN 141 on PDC 1 (outside MPR 13[+/-128])
DBG: [PDS] PDC 1 [Rx 1] [PSN 142] [RUD_REQ/REQ_STD] - RX PACKET (1124)
```
and hangs.

**Comment by insanum** [comment]
No doubt bugs like this exist in the current implementation. 🙃

Can you provide additional debug logs from both the client and server side that show the bitmap state before these retransmit logs? I implemented the PDS to be quite verbose for exactly this types of situations.

**Comment by svyatoslavnik** [comment]
Sure @insanum 

```
cd uet-ref-prov
make clean
git checkout main
make
sudo su
export LD_LIBRARY_PATH=../libfabric/src/.libs
export UET_PDS=pds
export UET_IFNAME=enp2s0
./uet server test <client_ip_address> or ./uet client test <server_ip_address> 
```

[client.log](https://github.com/user-attachments/files/18011980/client.log)
[server.log](https://github.com/user-attachments/files/18011981/server.log)

**Comment by svyatoslavnik** [comment]
Hi @insanum!

Seems like we rooted this out. As soon as it passes internal review we will do a PR with proper description. And sure we will add a description into this Issue as well.

**Comment by svyatoslavnik** [comment]
Fix addressed in #52
